docs/template_plugin/tests/functional/op_reference/topk.cpp

```diff
@@ -0,0 +1,1189 @@
+            Tensor(ET_OUT, {3}, std::vector<T_OUT>{0, 3, 4}),
+            "topk_mode_sort_order_1"),
+
+        TopKParams(
+            Tensor(ET, {5}, std::vector<T>{3, 1, 2, 5, 4}),
+            Tensor(ET2, {}, std::vector<T2>{3}),
+            0,
+            op::v1::TopK::Mode::MIN,
+            op::v1::TopK::SortType::SORT_VALUES,
+            Tensor(ET, {3}, std::vector<T>{1, 2, 3}),
+            Tensor(ET_OUT, {3}, std::vector<T_OUT>{1, 2, 0}),
+            "topk_mode_sort_order_2"),
+
+        TopKParams(
+            Tensor(ET, {5}, std::vector<T>{3, 1, 2, 5, 4}),
+            Tensor(ET2, {}, std::vector<T2>{3}),
+            0,
+            op::v1::TopK::Mode::MIN,
+            op::v1::TopK::SortType::SORT_INDICES,
+            Tensor(ET, {3}, std::vector<T>{3, 1, 2}),
+            Tensor(ET_OUT, {3}, std::vector<T_OUT>{0, 1, 2}),
+            "topk_mode_sort_order_3"),
+    };
+    return params;
+}
+
+std::vector<TopKParams> generateCombinedParamsV3() {
+    const std::vector<std::vector<TopKParams>> generatedParams {
+        generateParamsMaxMinSort<element::Type_t::i8, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::i16, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::i32, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::i64, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::u8, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::u16, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::u32, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::u64, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::bf16, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::f16, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::f32, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::f64, element::Type_t::i64, element::Type_t::i32>(),
+    };
+    std::vector<TopKParams> combinedParams;
+
+    for (const auto& params : generatedParams) {
+        combinedParams.insert(combinedParams.end(), params.begin(), params.end());
+    }
+    return combinedParams;
+}
+
+INSTANTIATE_TEST_SUITE_P(smoke_TopK_With_Hardcoded_Refs, ReferenceTopKTestV3,
+    testing::ValuesIn(generateCombinedParamsV3()), ReferenceTopKTest::getTestCaseName);
+
+class ReferenceTopKTest1d : public ReferenceTopKTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.A.data};
+        refOutData = {params.result0.data};
+    }
+
+    void SetUp1() {
```

Comment:
Very strange name. Can we use more clear name?

Reply:
It is updated to SetUpSecond.